File: app/src/main/java/forpdateam/ru/forpda/settings/SwitchPreference.java
package forpdateam.ru.forpda.settings;

import android.content.Context;
import android.os.Build;
import android.support.annotation.RequiresApi;
import android.support.v7.preference.SwitchPreferenceCompat;
import android.util.AttributeSet;

/**
 * Created by radiationx on 26.07.17.
 */

/*
* Исправляет самопроизвольные переключения настроек в киткате.
* Пи*дец, да.
* */
public class SwitchPreference extends SwitchPreferenceCompat {
    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    public SwitchPreference(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    public SwitchPreference(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public SwitchPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SwitchPreference(Context context) {
        super(context);
    }
}

File: app/src/main/java/forpdateam/ru/forpda/ui/views/SwitchPreference.kt
package forpdateam.ru.forpda.ui.views

import android.content.Context
import android.util.AttributeSet
import androidx.preference.SwitchPreferenceCompat

/**
 * Created by radiationx on 26.07.17.
 */
/*
 * Исправляет самопроизвольные переключения настроек в киткате.
 * Пи*дец, да.
 * */
class SwitchPreference : SwitchPreferenceCompat {
    constructor(
        context: Context,
        attrs: AttributeSet?,
        defStyleAttr: Int,
        defStyleRes: Int
    ) : super(context, attrs, defStyleAttr, defStyleRes)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context) : super(context)
}
